frame/election-provider-multi-phase/src/weights.rs

```diff
@@ -68,45 +70,45 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
 	// Storage: Staking ForceEra (r:1 w:0)
 	// Storage: ElectionProviderMultiPhase CurrentPhase (r:1 w:0)
 	fn on_initialize_nothing() -> Weight {
-		Weight::from_ref_time(13_495_000 as u64)
+		Weight::from_ref_time(17_624_000 as u64)
 			.saturating_add(T::DbWeight::get().reads(8 as u64))
 	}
 	// Storage: ElectionProviderMultiPhase Round (r:1 w:0)
 	// Storage: ElectionProviderMultiPhase CurrentPhase (r:0 w:1)
 	fn on_initialize_open_signed() -> Weight {
-		Weight::from_ref_time(14_114_000 as u64)
+		Weight::from_ref_time(17_260_000 as u64)
 			.saturating_add(T::DbWeight::get().reads(1 as u64))
 			.saturating_add(T::DbWeight::get().writes(1 as u64))
 	}
 	// Storage: ElectionProviderMultiPhase Round (r:1 w:0)
 	// Storage: ElectionProviderMultiPhase CurrentPhase (r:0 w:1)
 	fn on_initialize_open_unsigned() -> Weight {
-		Weight::from_ref_time(13_756_000 as u64)
+		Weight::from_ref_time(17_307_000 as u64)
 			.saturating_add(T::DbWeight::get().reads(1 as u64))
 			.saturating_add(T::DbWeight::get().writes(1 as u64))
 	}
 	// Storage: System Account (r:1 w:1)
 	// Storage: ElectionProviderMultiPhase QueuedSolution (r:0 w:1)
 	fn finalize_signed_phase_accept_solution() -> Weight {
-		Weight::from_ref_time(28_467_000 as u64)
+		Weight::from_ref_time(34_995_000 as u64)
 			.saturating_add(T::DbWeight::get().reads(1 as u64))
 			.saturating_add(T::DbWeight::get().writes(2 as u64))
 	}
 	// Storage: System Account (r:1 w:1)
 	fn finalize_signed_phase_reject_solution() -> Weight {
-		Weight::from_ref_time(21_991_000 as u64)
+		Weight::from_ref_time(27_568_000 as u64)
 			.saturating_add(T::DbWeight::get().reads(1 as u64))
 			.saturating_add(T::DbWeight::get().writes(1 as u64))
 	}
 	// Storage: ElectionProviderMultiPhase SnapshotMetadata (r:0 w:1)
 	// Storage: ElectionProviderMultiPhase DesiredTargets (r:0 w:1)
 	// Storage: ElectionProviderMultiPhase Snapshot (r:0 w:1)
-	fn create_snapshot_internal(v: u32, t: u32, ) -> Weight {
-		Weight::from_ref_time(3_186_000 as u64)
-			// Standard Error: 1_000
-			.saturating_add(Weight::from_ref_time(202_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 3_000
-			.saturating_add(Weight::from_ref_time(60_000 as u64).saturating_mul(t as u64))
```

Comment:
big diff

Reply:
I still haven't finished analyzing this one, but what's interesting is that:

1) when generated on my machine this benchmark has a base weight of zero when running without my PR (so it is affected by the zero weights issue),
2) when I take all of the raw `((v, t), time)` datapoints, plug them into both weight formulas, and compare the resulting numbers with the raw extrinsic times then I get (in general) more accurate predictions of execution time with the new approach: total prediction error is half of what it was, and the average error is 67% of what it was. Although please note this is *with* the 'before' base weight of zero, since that's what is generated on my machine, so it is a little unfair.

Anyway, indeed the base weight here is too large, mainly due to the minimum values of all of the components being huge. If I change them to be at minimum `1` instead of `1000` and `500` then the base weight drops down to a more reasonable ~200M instead of the current ~500M.